.github/workflows/release.yml

```diff
@@ -12,15 +12,152 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-name: release
+name: Release Images and Charts
 
 on:
   release:
     types:
       - 'created'
 
 jobs:
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Shellcheck
+        uses: reviewdog/action-shellcheck@v1
+        with:
+          github_token: ${{ secrets.github_token }}
+          reporter: github-pr-review
+          path: "."
+          pattern: "*.sh"
+
+      - uses: cachix/install-nix-action@v22
+      - uses: rrbutani/use-nix-shell-action@v1.1.0
+        with:
+          file: shell.nix
+
+      - name: Set up Go 1.19
+        uses: actions/setup-go@v5
+        with:
+          go-version: 1.19.13
+          cache: false
+
+      - name: Lint Check
+        uses: golangci/golangci-lint-action@v4
+        with:
+          version: v1.56.2
+          args: -E exportloopref,dupl,revive,bodyclose,goconst,misspell -D structcheck --timeout 5m0s
+
+      - name: Check if the chart follows a valid semver version
+        run: |
+          branch_name="${{ github.event.pull_request.base.ref }}"
+          ./scripts/validate-chart-version.sh --branch $branch_name
+
+      - name: Check if the chart is publishable
+        run: |
+          TAG=${{ github.event.release.tag_name }}
+          ./scripts/update-chart-version.sh --tag $TAG --publish-release
+
+      - name: Run chart-testing lint
+        run: |
+          ct lint --config ct.yaml
+
+  unit-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go 1.19
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.19.13
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Unit test
+        run: make test
+
+      - name: Verify corrections
+        run: make verify-src
+
+      - name: Upload Coverage Report
+        uses: codecov/codecov-action@v4
+
   provisioner-localpv:
+    runs-on: ubuntu-latest
+    needs: ['lint', 'unit-test']
+    steps:
+      - name: Set up Go 1.19
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.19.13
+
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: all
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+        with:
+          version: v0.5.1
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Build multi-arch Image
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          file: ./buildscripts/provisioner-localpv/provisioner-localpv.Dockerfile
+          push: false
+          load: false
+          platforms: linux/amd64, linux/arm64, linux/arm/v7, linux/ppc64le
+          tags: |
+            openebs/provisioner-localpv:ci
```

Comment:
IIRC, we're moving away from the 'ci' tag. Does this need changes?

Reply:
Ahh missed this. This seems to be incorrect. The ci tag is no longer needed.

Reply:
Done.